pylint/checkers/typecheck.py

```diff
@@ -1420,8 +1420,15 @@ def _check_argument_order(
             self.add_message("arguments-out-of-order", node=node, args=())
 
     def _check_isinstance_args(self, node: nodes.Call) -> None:
-        if len(node.args) != 2:
-            # isinstance called with wrong number of args
+        if len(node.args) > 2:
+            # for when isinstance called with too many args
+            self.add_message(
+                "too-many-arguments",
+                node=node,
+                args=(len(node.args), 2),
+            )
```

Comment:
```suggestion
                confidence=HIGH,
            )
```

Reply:
Technically this check uses inference at the top of `visit_call`, but I'm wondering whether it's better to just say HIGH anyway.

Reply:
Could you really briefly explain the difference here between `HIGH` and `INFERENCE`?

Based on the docstring, it looks like "called functions/methods are inferred to callable objects" with `safe_infer`, so in this case, astroid is inferring `isinstance`'s arguments. Since we end up checking isinstance's inferred args, then I can see how it could make sense to have it be `INFERENCE`.  Then again, I noticed further below that the `kwarg-superseded-by-positional-arg` message comes with a `confidence=HIGH` designation.

I'm happy to add `HIGH` here, especially considering I'm very new to this codebase (though I do use pylint itself a decent amount), just trying to get a feel for the rationale.

Thanks!

Reply:
No problem. If a check relies on inference, we want to declare that with the confidence level INFERENCE, otherwise we use HIGH. This is so that pylint can be run with a confidence level to filter out messages in highly dynamic code bases where inference is often opaque.

My thought was that inference is probably being used here mostly for the argument types, not the call itself. We only need to use inference to find out if a call is to isinstance if it's aliased somehow, like:

```py
aliased = isinference
result = aliased(1, str)  # need inference to find out what `aliased` is
```

I would have a look and find out if that's really what we're doing in this check. If not, I think we can safely say HIGH.

> Since we end up checking isinstance's inferred args,

`isinstance-second-argument-not-valid-type` might be checking the inferred args, but for `too-many-arguments` we're just counting, not inferring.

Reply:
I see. Thanks for explaining!